mm2src/mm2_rpc/src/data/legacy/orders.rs

```diff
@@ -95,106 +197,309 @@ impl Default for MatchBy {
+pub struct OrdersHistoryRequest {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub order_type: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub initial_action: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub base: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub rel: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub from_price: Option<MmNumber>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub to_price: Option<MmNumber>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub from_volume: Option<MmNumber>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub to_volume: Option<MmNumber>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub from_timestamp: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub to_timestamp: Option<u64>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub was_taker: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub status: Option<String>,
+    pub include_details: bool,
 }
 
-construct_detailed!(AggregatedBaseVol, base_max_volume_aggr);
-construct_detailed!(AggregatedRelVol, rel_max_volume_aggr);
+#[derive(Serialize, Deserialize)]
+pub struct OrdersHistoryResponse {
+    pub orders: Vec<FilteringOrder>,
+    pub details: Vec<OrderForRpc>,
+    pub found_records: usize,
+    pub warnings: Vec<UuidParseError>,
+}
 
-#[derive(Clone, Default, Debug, Eq, PartialEq, Serialize, Deserialize)]
-pub struct OrderConfirmationsSettings {
-    pub base_confs: u64,
-    pub base_nota: bool,
-    pub rel_confs: u64,
-    pub rel_nota: bool,
+#[derive(Debug, Serialize, Deserialize)]
+pub struct FilteringOrder {
+    pub uuid: String,
+    pub order_type: String,
+    pub initial_action: String,
+    pub base: String,
+    pub rel: String,
+    pub price: f64,
+    pub volume: f64,
+    pub created_at: i64,
+    pub last_updated: i64,
+    pub was_taker: i8,
+    pub status: String,
 }
 
-impl OrderConfirmationsSettings {
-    pub fn reversed(&self) -> OrderConfirmationsSettings {
-        OrderConfirmationsSettings {
-            base_confs: self.rel_confs,
-            base_nota: self.rel_nota,
-            rel_confs: self.base_confs,
-            rel_nota: self.base_nota,
-        }
-    }
+#[derive(Serialize, Deserialize)]
+pub struct UuidParseError {
+    pub uuid: String,
+    pub warning: String,
 }
+
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "method", rename = "update_maker_order")]
+pub struct UpdateMakerOrderRequest {
+    pub uuid: Uuid,
+    pub new_price: Option<MmNumber>,
+    pub max: Option<bool>,
+    pub volume_delta: Option<MmNumber>,
+    pub min_volume: Option<MmNumber>,
+    pub base_confs: Option<u64>,
+    pub base_nota: Option<bool>,
+    pub rel_confs: Option<u64>,
+    pub rel_nota: Option<bool>,
```

Comment:
Why does these fields don't require `[serde(skip_serializing_if = "Option::is_none")]`? There are other fields like that in other structs, I don't get why optional fields is sometimes require that and sometimes not.

Reply:
Thank you )
[done](https://github.com/KomodoPlatform/komodo-defi-framework/pull/1932/commits/001abfffbf55e7e82e4c1ce20b9b0083c528d780)